sqlmesh/core/engine_adapter/redshift.py

```diff
@@ -128,7 +128,16 @@ def _fetch_native_df(
     ) -> pd.DataFrame:
         """Fetches a Pandas DataFrame from the cursor"""
         self.execute(query, quote_identifiers=quote_identifiers)
-        return self.cursor.fetch_dataframe()
+
+        # We manually build the `DataFrame` here because the driver's `fetch_dataframe`
+        # method does not respect the active case-sensitivity configuration.
+        #
+        # Context: https://github.com/aws/amazon-redshift-python-driver/issues/238
+        fetcheddata = self.cursor.fetchall()
+        columns = [column[0] for column in self.cursor._getDescription() or []]
```

Comment:
Just curious why not just use `self.cusror.description`? 
https://github.com/aws/amazon-redshift-python-driver/blob/master/redshift_connector/cursor.py#L526

Reply:
Just a random brain fart, I will amend. Thanks for pointing it out!